pkg/graveler/errors.go

```diff
@@ -59,6 +59,8 @@ var (
 	ErrTooManyTries                 = errors.New("too many tries")
 	ErrSkipValueUpdate              = errors.New("skip value update")
 	ErrImport                       = wrapError(ErrUserVisible, "import error")
+	ErrReadOnlyRepository           = errors.New("read-only repository")
+	ErrWriteToReadOnlyRepository    = wrapError(ErrReadOnlyRepository, "cannot write to a read-only repository")
```

Comment:
Agree with @guy-har don't think we need to have two error types if one is visible we can use it. Prefer to report that the repository is read-only without specific operation. so we can report the error on delete branch for example.